File: python/ex012.py
# Faça uma algoritimo que leia o preço de um produto e mostre se unovo preço, com 5% de desconto.
while True:
    entry = input("Product price: ")
    try: 
        product_price = float(entry)
        break
    except ValueError:
        print("Please, type a valid number")

five_percent_of_discount = (product_price * 5) / 100
product_with_discount = product_price - five_percent_of_discount

print("Your product with 5% of discount is R${:.2f}".format(product_with_discount))

File: rust/ex012.rs
use std::io;

fn main() {
    let product_price = get_product_price();
    let five_percent_of_discount = (product_price * 5.0) / 100.00;
    let product_with_discount = product_price - five_percent_of_discount;
    println!("O seu produto no valor de {} com 5% de desconto é R${:.2}", product_price, product_with_discount);
}

fn get_product_price() -> f32 {
    loop {
        let mut entry = String::new();
        println!("Preço do produto: ");
        io::stdin()
            .read_line(&mut entry)
            .expect("Error reading line.");

        match entry.trim().parse::<f32>() {
            Ok(product_price) => return product_price,
            Err(_) => {
                println!("Por favor, digite um número válido.")
            }
        }
    }
}
